content/Projects/Observability/kube-rbac-proxy.md

```diff
@@ -0,0 +1,136 @@
+# kube-rbac-proxy
+
+`kube-rbac-proxy`, as the name suggests, is a proxy that sits in front of a workload and performs authentication and authorization on its behalf, to make sure the requests passing through are the ones that it expects to.
+
+## Workflow
+
+The purpose of `kube-rbac-proxy` is to distinguish between calls made by same or different user(s) (or service account(s)) to endpoint(s) and protect them from unauthorized resource access based on their trusted identity (e.g. tokens, TLS certificates, etc.) or the RBACs they hold, respectively.
+
+- [**Authentication**](https://github.com/brancz/kube-rbac-proxy/blob/74181c75b8b6fbcde7eff1ca5fae353faac5cfae/pkg/authn/config.go#L33-L39): Both [OIDC JWT](https://github.com/brancz/kube-rbac-proxy/blob/52e49fbdb75e009db4d02e3986e51fdba0526378/pkg/authn/oidc.go#L45-L63) and [delegated (header-based)](https://github.com/kubernetes/apiserver/blob/master/pkg/authentication/authenticatorfactory/delegating.go#L102-L112) authentication rely on Bearer tokens to authenticate the requests. The tokens are validated against the OIDC provider or the Kubernetes API server, respectively. Monitoring components use [mTLS](#downstream-usage) for client-server authentication, which provides a secure way to establish trust based on each other's digital certificates. The proxy can also possibly be configured to [skip the authentication step](https://github.com/brancz/kube-rbac-proxy/blob/4c4a18fa7dd5d8069c0aa3a55028e06631621a52/pkg/authn/delegating.go#L53) altogether, in which case it will not validate the tokens and will assume that the requests are authenticated, which is not recommended. This token represents the identity of the user or service account that is making the request. A [`TokenReview` request is created](https://github.com/kubernetes/apiserver/blob/21bbcb57c672531fe8c431e1035405f9a4b061de/plugin/pkg/authenticator/token/webhook/webhook.go#L51-L53), and includes the bearer token that the client provided, which is then sent to the API server. The API server verifies the authenticity of the token and responds with the details of the authenticated party, if the operation was successful.
```

Comment:
You don't mention TLS based authentication?

Reply:
I wasn't sure since we mention `mTLS` which is and extension of TLS, since in the former case both client and server validate their certs, as compared to the latter's client-side validation.